### PR TITLE
Add TLS configuration options

### LINKED ISSUE #698: Unrecognized option "tlsCAFile"
I am trying to connect to an AWS DocumentDB with TLS enabled. I see no way passing the ssl/tls caFile option.

I assume I should be able to configure this somehow like this:

```yaml
options:
    username: '%env(resolve:MONGODB_USERNAME)%'
    password: '%env(resolve:MONGODB_PASSWORD)%'
    ssl: '%env(bool:MONGODB_SSL)%'
    tlsCAFile: '/path/to/cafile.pem'
```

Are there any plans to add this options or reasons to not do so?

### PR DESCRIPTION
**Pull request by hasc**
Add TLS configuration options as described in [MongoDB\Driver\Manager::__construct](https://www.php.net/manual/en/mongodb-driver-manager.construct.php)

### CODE REVIEW

File: Resources/config/schema/mongodb-1.0.xsd [hunk 1]
```diff
@@ -68,6 +68,15 @@
     <xsd:attribute name="readPreference" type="read-preference" />
     <xsd:attribute name="replicaSet" type="xsd:string" />
     <xsd:attribute name="ssl" type="xsd:boolean" />
+    <xsd:attribute name="tls" type="xsd:string" />
```
Comment (franmomu) [inline Resources/config/schema/mongodb-1.0.xsd]:
```suggestion
    <xsd:attribute name="tls" type="xsd:boolean" />
```

**Review by franmomu — COMMENTED** [review]
Thanks @hasc for submitting the PR!

File: Resources/doc/config.rst [hunk 2]
```diff
@@ -576,21 +576,30 @@ Full Default Configuration
...
+                        replicaSet:                             ~ # replica set name
+                        socketTimeoutMS:                        ~
+                        ssl:                                    ~
+                        tls:                                    ~
+                        tlsAllowInvalidCertificates:             ~
+                        tlsAllowInvalidHostnames:               ~
+                        tlsCAFile:                              ~
+                        tlsCertificateKeyFile:                   ~
+                        tlsCertificateKeyFilePassword:           ~
+                        tlsDisableCertificateRevocationCheck:    ~
```
Comment (malarzm) [inline Resources/doc/config.rst]: nitpicking, but let's keep those `~`aligned here

File: Tests/DependencyInjection/ConfigurationTest.php [hunk 3]
```diff
@@ -99,15 +99,24 @@ public function testFullConfiguration(array $config): void
...
+                        'replicaSet'                          => 'foo',
+                        'slaveOkay'                           => true,
+                        'socketTimeoutMS'                     => 1000,
+                        'ssl'                                 => true,
+                        'tls'                                 => true,
+                        'tlsAllowInvalidCertificates'          => false,
+                        'tlsAllowInvalidHostnames'            => false,
+                        'tlsCAFile'                           => '/path/to/cert.pem',
+                        'tlsCertificateKeyFile'                => '/path/to/key.crt',
+                        'tlsCertificateKeyFilePassword'        => 'secret',
```
Comment (malarzm) [inline Tests/DependencyInjection/ConfigurationTest.php:111]: and those `=>` here

File: Tests/DependencyInjection/Fixtures/config/yml/full.yml [hunk 4]
```diff
@@ -42,6 +42,15 @@ doctrine_mongodb:
...
                 ssl:              true
+                tls:                                 true
+                tlsAllowInvalidCertificates:          false
+                tlsAllowInvalidHostnames:            false
+                tlsCAFile:                           '/path/to/cert.pem'
+                tlsCertificateKeyFile:                '/path/to/key.crt'
+                tlsCertificateKeyFilePassword:        'secret'
+                tlsDisableCertificateRevocationCheck: false
+                tlsDisableOCSPEndpointCheck:         false
+                tlsInsecure:                         false
```
Comment (malarzm) [inline Tests/DependencyInjection/Fixtures/config/yml/full.yml]: and those values here (at least one to another, a space off here and there really itches ;)

**Review by malarzm — APPROVED** [review]

**Comment by alcaeus** [comment]
Instead of piecemeal additions, I'd rather not have to duplicate all options from the extension here. IMO, we should solicit the options and driverOptions as arrays and let the MongoDB extension handle them entirely.

**Comment by malarzm** [comment]
> Instead of piecemeal additions, I'd rather not have to duplicate all options from the extension here. IMO, we should solicit the options and driverOptions as arrays and let the MongoDB extension handle them entirely.

That's a nice idea for us, but IMO a bit worse for developers using the bundle:
- autocompletion is lost since we don't provide keys so IDEs can't pick that up
- not sure how extension handles typos in keys, but Symfony would let you know right away (and what probably you should write in the first place)

File: Tests/DependencyInjection/ConfigurationTest.php (hunk 3 above)
Comment (hasc) [reply Tests/DependencyInjection/ConfigurationTest.php:111]: Done

File: Tests/DependencyInjection/Fixtures/config/yml/full.yml (hunk 4 above)
Comment (hasc) [reply Tests/DependencyInjection/Fixtures/config/yml/full.yml]: Done

File: Resources/doc/config.rst (hunk 2 above)
Comment (hasc) [reply Resources/doc/config.rst]: Done

**Comment by malarzm** [comment]
@hasc thanks for the updates! Could you please squash your commits so there's just 1 instead of 5?

**Comment by hasc** [comment]
Sorry, something went wrong here. Git log looked good in the forked project. However my commit interferes with your commits. I need to figure out how to fix that. Will do so asap. Alternatively I could just create a new PR.

**Comment by hasc** [comment]
Finally, squashed to 1 commit

**Comment by hasc** [comment]
Sorry for the inconvenience. I've added another commit on top that also adds the configuration options "retryReads" and "retryWrites". I hope that's fine.

File: DependencyInjection/Configuration.php [hunk 5]
```diff
@@ -305,9 +305,20 @@ private function addConnectionsSection(ArrayNodeDefinition $rootNode)
...
+                                    ->scalarNode('tlsCertificateKeyFile')->end()
+                                    ->scalarNode('tlsCertificateKeyFilePassword')->end()
+                                    ->booleanNode('tlsDisableCertificateRevocationCheck')->end()
+                                    ->booleanNode('tlsDisableOCSPEndpointCheck')->end()
+                                    ->booleanNode('tlsInsecure')->end()
                                     ->scalarNode('username')
                                         ->validate()->ifNull()->thenUnset()->end()
                                     ->end()
+                                    ->scalarNode('retryReads')->end()
+                                    ->scalarNode('retryWrites')->end()
```
Comment (franmomu) [inline DependencyInjection/Configuration.php]: Based on the docs and usage:
```suggestion
                                    ->booleanNode('retryReads')->end()
                                    ->booleanNode('retryWrites')->end()
```

File: Resources/config/schema/mongodb-1.0.xsd [hunk 6]
```diff
@@ -68,8 +68,19 @@
...
+    <xsd:attribute name="tlsCAFile" type="xsd:string" />
+    <xsd:attribute name="tlsCertificateKeyFile" type="xsd:string" />
+    <xsd:attribute name="tlsCertificateKeyFilePassword" type="xsd:string" />
+    <xsd:attribute name="tlsDisableCertificateRevocationCheck" type="xsd:boolean" />
+    <xsd:attribute name="tlsDisableOCSPEndpointCheck" type="xsd:boolean" />
+    <xsd:attribute name="tlsInsecure" type="xsd:boolean" />
     <xsd:attribute name="socketTimeoutMS" type="xsd:integer" />
     <xsd:attribute name="username" type="xsd:string" />
+    <xsd:attribute name="retryReads" type="xsd:string" />
+    <xsd:attribute name="retryWrites" type="xsd:string" />
```
Comment (franmomu) [inline Resources/config/schema/mongodb-1.0.xsd]:
```suggestion
    <xsd:attribute name="retryReads" type="xsd:boolean" />
    <xsd:attribute name="retryWrites" type="xsd:boolean" />
```

File: DependencyInjection/Configuration.php (hunk 5 above)
Comment (hasc) [reply DependencyInjection/Configuration.php]: Sorry, that was sloppy. Fixed and squashed

File: Resources/config/schema/mongodb-1.0.xsd (hunk 6 above)
Comment (hasc) [reply Resources/config/schema/mongodb-1.0.xsd]: Sorry, that was sloppy. Fixed and squashed

**Review by malarzm — APPROVED** [review]

**Review by franmomu — APPROVED** [review]

**Comment by malarzm** [comment]
Thanks @hasc!

**Pull request merged**